src/hypothesis/strategies.py

```diff
@@ -1575,4 +1578,95 @@ def deferred(definition):
     return DeferredStrategy(definition)
 
 
+class _PathLike(object):
+    """A class implementing the os.PathLike protocol available since
+    Python 3.6.
+
+    See https://www.python.org/dev/peps/pep-0519/ for more details.
+    """
+
+    def __init__(self, value):
+        self._value = value
+
+    def __fspath__(self):
+        return self._value
+
+    def __repr__(self):
+        return "pathlike(%r)" % self._value
+
+
+@defines_strategy
+@composite
+def fspaths(draw, allow_pathlike=True):
+    """A strategy which generates filesystem path values.
```

Comment:
The more I think about the chance of test code operating on paths, the more worried I get - if some downstream user writes a test that opens example paths in write mode, the results will be nasty.

We definitely need a warning that the resulting paths should not be opened (on pain of `atime` disruption in read-only mode, or debris and clobbering in write-mode).  I'd also consider adding an argument `allow_existing=False`, to exclude paths which already exist from being generated.